subprojects/packagefiles/openssl/meson.build.tmpl

```diff
@@ -0,0 +1,88 @@
+# OpenSSL library
+libcrypto_sources = [
+%%-
+foreach $src (@libcrypto_srcs) {
+  $OUT .= "  '$src',\n";
+}
+foreach $src (@generated_srcs) {
+  $OUT .= "  'generated-config/archs/$arch/$asm/$src',\n";
+}
+-%%]
+libssl_sources = [
+%%-
+foreach $src (@libssl_srcs) {
+  $OUT .= "  '$src',\n";
+}
+-%%]
+openssl_defines = [
+%%-
+foreach $define (@{$config{defines}}) {
+  $OUT .= "  '$define',\n";
```

Comment:
We're ending up with every single generated-config/ defining NDEBUG, but project default options can declare `'b_ndebug=true'` once and for all.

`'b_ndebug=if-release'` would be a better fit, since openssl's ./Configure just adds a global -DNDEBUG if the buildtype is release. :D

Reply:
Interesting, it does work when specified in CLI, but in project's `default_options` `b_ndebug` doesn't seem to have any effect even when set to `true`. I'm checking `build.ninja` file.

Reply:
Ugh, this may just not be available per-subproject...

Reply:
I removed unconditional define though and optimization level with it since it is derived from build type anyway